### PR TITLE
Refactor Makefile and update local dev options

### PR DESCRIPTION
**Pull request by youngnick**
This is a WIP PR to update the local dev options. I'm aiming to have
a make target that will:
- spin you up a kind cluster
- set you up with Envoys running in there, trying to connect to a Contour
  running on the host.
- with the SSL pulled out for you.

First up though, is a Makefile refactor to make it a bit more composable.

This is a WIP PR because it's quicker to use Actions for testing than my
local machine. :)

This commit splits up many of the make targets into sub-steps,
that we can reuse, instead of having the same script called in
multiple places.

I've also removed the old local-dev shortcuts, I think we can do
better.

The intent here is to make it so that we can use the pieces
more easily, eventually aiming at making local dev and bug
reproduction quicker and easier.

Signed-off-by: Nick Young <ynick@vmware.com>

### CODE REVIEW

**Review by sunjayBhatia — COMMENTED** [review]
the Makefile refactor looks good to me, pulled the branch down and messed with it 👍🏽 

are you intending to add the local dev commands to this PR as well?

File: Makefile [hunk 1]
```diff
@@ -331,28 +313,51 @@ site-devel: ## Launch the website
 site-check: ## Test the site's links
 	# TODO: Clean up to use htmltest
 
-.PHONY: e2e
-e2e: ## Run E2E tests against a real k8s cluster
+
+# Tools for testing and troubleshooting
+
+.PHONY: make-kind-cluster
+make-kind-cluster: ## Make a kind cluster with standard ports forwarded
```
[the remark this answers is not in the document]
Comment (youngnick) [reply Makefile:320]: haha, that is a great suggestion. I'll update.

**Comment by youngnick** [comment]
Yes, I'm working on the local dev commands here too, but maybe I should just push this in first. I'll tidy it up and do that, make the PRs easier to review.

**Comment by youngnick** [comment]
This should be good for actual review now. I'll build the local-dev stuff from here.

**Review by sunjayBhatia — APPROVED** [review]

File: Makefile [hunk 2]
```diff
@@ -338,28 +320,51 @@ site-devel: ## Launch the website
...
+.PHONY: cleanup-kind
+cleanup-kind:
 	./test/scripts/cleanup.sh
 
-.PHONY: upgrade
-upgrade: ## Run upgrade tests against a real k8s cluster
-	./test/scripts/make-kind-cluster.sh
-	./test/scripts/install-contour-release.sh $(CONTOUR_UPGRADE_FROM_VERSION)
+.PHONY: load-contour-image-kind
+load-contour-image-kind:
```
Comment (stevesloka) [inline Makefile:350]: Can probably remove this since it's required to have the image exported first to disk at a specific place.

Comment (stevesloka) [reply Makefile:350]: Or if you want to keep, add a comment stating what needs set to properly use?

**Review by stevesloka — CHANGES_REQUESTED** [review]

File: Makefile [hunk 3]
```diff
@@ -46,6 +46,8 @@ BUILD_GOPROXY ?= https://proxy.golang.org
 # LOCALIP as an env var before running 'make local' will solve that.
 LOCALIP ?= $(shell ifconfig | grep inet | grep -v '::' | grep -v 127.0.0.1 | head -n1 | awk '{print $$2}')
 
+DOCKERIP ?= $(shell docker network inspect kind | jq -r ".[0].IPAM.Config[0].Gateway")
```
Comment (stevesloka) [inline Makefile:49]: This isn't used anywhere in this changeset?

**Comment by stevesloka** [comment]
Added label `release-note` due to the deprecation comment.

> This method of certificate generation is DEPRECATED and will be removed soon.

File: Makefile (hunk 3 above)
Comment (youngnick) [reply Makefile:49]: ah, thanks, I'll pull it out. It will be in the next one instead.

File: Makefile (hunk 2 above)
Comment (youngnick) [reply Makefile:350]: I was just moving things around, but I'll have a look and see if we can use Make properly here, and have a rule that will rebuild the contour image if stuff has changed, which can then be a prereq for this rule.

Comment (youngnick) [reply Makefile:350]: Agreed that if I keep, I'll add a comment which tells you what this step expects.

**Review by skriss — COMMENTED** [review]
LGTM pending open comments

File: Makefile (hunk 2 above)
Comment (youngnick) [reply Makefile:350]: I've updated with an explanatory comment and a `make help` line.

**Comment by youngnick** [comment]
Okay, actually resolved the pending comments this time, thanks everyone. Should be good for final review now.

**Review by skriss — APPROVED** [review]
LGTM

**Pull request merged**